feat: add support for Seyond LiDAR driver

citruscosmos opened the pull request:
### **Description**

This pull request introduces support for the official `seyond_ros_driver` for Seyond LiDARs, adding it as an alternative to the existing `nebula_ros` driver.

To ensure backward compatibility and avoid breaking existing configurations, this new driver is **not enabled by default**. A new launch argument, `lidar_driver_type`, has been added to select the desired driver. The default value for this argument is set to `nebula`.

### **Key Changes**

* **New Launch Argument**:
    * Introduced a new `lidar_driver_type` argument in `drs.launch.xml`.
    * This argument defaults to `nebula` to maintain existing behavior.
* **Dynamic Driver Launching**:
    * The `drs_ecu0.launch.xml` and `drs_ecu1.launch.xml` files have been updated to dynamically load the component launch file based on the `lidar_driver_type` variable (e.g., `drs_$(var lidar_driver_type).launch.xml`).
* **New Dependencies**:
    * Added `tier4/seyond_ros_driver` and its dependency `Seyond-Inc/inno-lidar-sdk` to the `drs.repos` file.
    * Added `<exec_depend>seyond</exec_depend>` to the `drs_launch/package.xml` to declare the new package dependency.
* **Launch File Updates**:
    * Modified `drs_seyond.launch.xml` to launch the `seyond_component_node` from the `seyond` package instead of the previously used `seyond_ros_wrapper_node` from `nebula_ros`.
* **Dependency Fix**:
    * Updated the URL for the `inno-lidar-sdk` dependency to use `https` instead of `git@`.

### **Verification**
* Confirmed seyond LiDAR driver works correctly and can be recorded with HILS by updating `lidar_driver_type` to `seyond` and record topics.
* Seyond SDK is built within colcon build process.

citruscosmos committed:
feat: add seyond driver but not used by default

citruscosmos committed:
fix: update drs.repos for inno-lidar-sdk to use http instead of git

otenim committed:
Merge branch 'develop/r36.4.0' into feat/add-seyond-driver

otenim commented on drs.repos:
上でseyond_sdkをcloneしているのでもういらないです

otenim replied:
修正しました

otenim replied:
あいや、すいませんこれ必要ですね

citruscosmos replied:
修正ありがとうございます。

otenim committed:
refactor: no more needed to clone as its already cloned earlier

otenim commented:
@citruscosmos 
Confirm build successfully finish 
```bash
rm -rf src/bag_converter/src/seyond_decoder/src/seyond_sdk
colcon build --packages-up-to drs_launch
```

otenim commented:
ミスです。やっぱり必要ですねこれ。

otenim committed:
Revert "refactor: no more needed to clone as its already cloned earlier"

This reverts commit c0272bfc13108d71a8a3fb34d97ca939036daf90.

otenim commented on drs.repos:
```diff
@@ -15,6 +15,15 @@ repositories:
+  dependencies/seyond_ros_driver/src/seyond_lidar_ros/src/seyond_sdk:
+    type: git
+    url: https://github.com/Seyond-Inc/inno-lidar-sdk.git
+    version: e7e40ca635605447dbf8d73735706aee56b13f70  # main
```
@citruscosmos 
このコミットIDにした理由は何ですか?

citruscosmos replied:
seyond_ros_driverで、seyond_sdkをビルドして生成されるライブラリ(.aファイル)が必要となるためです。

上記のbag_converter/src/seyond_decoder/src/seyond_sdkをdrs.reposから削除したものを、クリーンビルドを実施してみましたが、問題なくビルドが通りました。

otenim committed:
chore: unify seyond_sdk version across packages

Align seyond_sdk commit ID in bag_converter to match seyond_ros_driver
(e7e40ca) to ensure consistency between both packages.
